celeryutils/persist_on_failure.py

```diff
@@ -0,0 +1,52 @@
+"""
+Celery utility code for persistent tasks.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from celery import Task
+
+from .models import FailedTask
+
+
+# pylint: disable=abstract-method
+class PersistOnFailureTask(Task):
+    """
+    Custom Celery Task base class that persists task data on failure.
+    """
+
+    abstract = True
```

Comment:
It's a celery flag to prevent this task from getting registered as a callable task.